### ISSUE TITLE
Websocket Support

### ISSUE THREAD

**Issue by finiteinfinity** [brefphp/bref#744]
Since AWS announced the support for [websockets via API Gateway](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-websocket-api.html) it appears that we could implement support for a using bref as a websocket handler as a lambda.

In terms of implementation, I'd imagine we could add another event for supporting invocation via API gateway?

Obviously if this isn't possible on the current framework then just close this issue!

**Comment by mnapoli** [comment]
Hi, yes this is indeed possible (we'd need to add a new event class and handler class too).

I would be very curious to know how well it works.

**Comment by finiteinfinity** [comment]
Thanks for the response @mnapoli glad to hear its possible. I'm going to start taking a look at how this could be implemented into existing frameworks such as Laravel or Symfony. Will update with comments on my findings

**Comment by danniehansen** [comment]
@finiteinfinity @mnapoli Found this interesting as I've got a few production services running using ApiGateway websockets + lambda(nodejs). Started work on a pull request to implement the typed ApiGateway websocket events. I've got it working and it's correctly parsing the events. What's left to do is to figure out if the implementation is solid, add tests and clean up the source code if need be.

https://github.com/brefphp/bref/pull/773

The actual stability of ApiGateway websockets + Lambda is really nice. Never had much of an issue.
